src/psij/executors/batch/template_function_library.py

```diff
@@ -0,0 +1,60 @@
+from datetime import timedelta
+from typing import Dict, Callable, Any, Union
+
+
+def walltime_to_minutes(walltime: Union[timedelta, int, str]) -> int:
+    """Converts a walltime object to a number of minutes.
+
+    The walltime can either be a Python :class:`~datetime.timedelta`, an integer, in which case
+    it is interpreted directly as a number of minutes, or a string with a format of either
+    `HH:MM:SS`, `HH:MM`, or `MM`.
+    Parameters
+    ----------
+    walltime
+        the walltime to convert
+    """
+    if isinstance(walltime, timedelta):
+        return int(walltime.total_seconds() / 60)
+    if isinstance(walltime, int):
+        return walltime
+    if isinstance(walltime, str):
+        return _parse_str_walltime(walltime)
+    raise ValueError('Invalid walltime: %s of type %s' % (walltime, type(walltime)))
+
+
+def _parse_str_walltime(walltime: str) -> int:
+    # accept HH:MM:SS or HH:MM or MM
+    els = walltime.split(':')
+    if len(els) > 3 or len(els) == 0:
```

Comment:
> `len(els) == 0`

can that even happen?

Reply:
For an empty string input, yes.

Reply:
I don't think so:
```py
$ python3
>>> ''.split(':')
['']
>>> len(''.split(':'))
1
>>> 
```

Reply:
Totally funny:

```Python
Python 3.8.10 (default, Jun  2 2021, 10:49:15)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> ''.split()
[]
>>> ''.split(':')
['']
>>>
```

Anyway...

Reply:
Yeah, I also dislike the special handling of a `None` separator, it is confusing... :-/